diff -Nru glchess-0.4.1/Makefile glchess-0.4.1-new/Makefile
--- glchess-0.4.1/Makefile	Fri Dec  7 19:51:54 2001
+++ glchess-0.4.1-new/Makefile	Sat Dec  8 02:13:50 2001
@@ -13,7 +13,7 @@
 	@cd src && make
 	@mv src/glchess .
 	@echo "Building documentation..."
-	@cat man/glchess.6 | sed -e s/VERSION/$(VERSION)/ -e s/DATE/$(DATE)/ | gzip -9 - > man/glchess.6.gz
+	@cat man/glchess.6 | sed -e s/VERSION/$(VERSION)/ -e s/DATE/$(DATE)/ > man/glchess.6
 
 # Updates version entry in src/global.h
 version:
