diff -Nru glchess-0.3.5/Makefile glchess-0.3.5-orig/Makefile
--- glchess-0.3.5/Makefile	Tue Sep  4 14:54:45 2001
+++ glchess-0.3.5-orig/Makefile	Sat Nov 10 21:34:28 2001
@@ -11,7 +11,7 @@
 	@cd src && make
 	@mv src/glchess .
 	@echo "Building documentation..."
-	@cat man/glchess.6 | sed -e s/VERSION/$(VERSION)/ -e s/DATE/$(DATE)/ | gzip -9 - > man/glchess.6.gz
+	@cat man/glchess.6 | sed -e s/VERSION/$(VERSION)/ -e s/DATE/$(DATE)/ > man/glchess.6.gz
 
 # Updates version entry in src/global.h
 version:
