diff -Nru glchess-0.4.2/configure.in glchess-0.4.2-new/configure.in
--- glchess-0.4.2/configure.in	Tue Dec 25 11:47:24 2001
+++ glchess-0.4.2-new/configure.in	Sat Dec 29 18:41:32 2001
@@ -75,7 +75,8 @@
 AC_C_CONST
 AC_TYPE_SIZE_T
 
-CFLAGS="$GTK_CFLAGS $GL_CFLAGS $GTKGL_CFLAGS"
+CC="$CC-FROM-RPM"
+CFLAGS="$CFLAGS-FROM-RPM $GTK_CFLAGS $GL_CFLAGS $GTKGL_CFLAGS"
 LIBS="$GTKGL_LIBS $GTK_LIBS $GL_LIBS $IMG_LIBS"
 
 AC_OUTPUT(src/Makefile man/Makefile textures/Makefile Makefile)
