diff -Nru glchess-0.4.1/src/Makefile glchess-0.4.1-new/src/Makefile
--- glchess-0.4.1/src/Makefile	Fri Dec  7 19:51:54 2001
+++ glchess-0.4.1-new/src/Makefile	Sat Dec  8 02:17:19 2001
@@ -1,5 +1,5 @@
-CC = gcc
-CFLAGS = -g -Wall `gtk-config --cflags`
+GCC = $(CC)
+CFLAGS-local = -g -Wall `gtk-config --cflags` $(CFLAGS)
 
 LIBS = `gtk-config --libs` -lgtkgl -lGL -lGLU -lm -L/usr/X11R6/lib -lXi -lXmu
 
@@ -23,7 +23,7 @@
 	texture.o
 
 all: $(OBJS)
-	$(CC) $(CFLAGS) $(OBJS) -o glchess $(LIBS)
+	$(GCC) $(CFLAGS-local) $(OBJS) -o glchess $(LIBS)
 	
 clean:
 	rm -f *.o *~ .\#* glchess
