diff -urN glchess-0.3.5/Makefile glchess-0.3.5-moje/Makefile
--- glchess-0.3.5/Makefile	Tue Sep  4 14:54:45 2001
+++ glchess-0.3.5-moje/Makefile	Thu Nov  8 16:43:50 2001
@@ -11,7 +11,7 @@
 	@cd src && make
 	@mv src/glchess .
 	@echo "Building documentation..."
-	@cat man/glchess.6 | sed -e s/VERSION/$(VERSION)/ -e s/DATE/$(DATE)/ | gzip -9 - > man/glchess.6.gz
+	@cat man/glchess.6 | sed -e s/VERSION/$(VERSION)/ -e s/DATE/$(DATE)/ | gzip -9nf - > man/glchess.6.gz
 
 # Updates version entry in src/global.h
 version:
diff -urN glchess-0.3.5/glchess.spec glchess-0.3.5-moje/glchess.spec
--- glchess-0.3.5/glchess.spec	Tue Sep  4 14:54:45 2001
+++ glchess-0.3.5-moje/glchess.spec	Thu Nov  8 22:08:21 2001
@@ -1,15 +1,20 @@
-Summary: glChess - A 3D chess interface
-Name: glchess
-Version: 0.3.5
-Release: 0
-Copyright: GPL
-Group: Amusements/Games
-BuildRoot: /var/tmp/%{name}-buildroot
-Source: %{name}-%{version}.tar.bz2
-Source1: glchess.menu
-Patch0: glchessrc.patch
-Packager: Giuseppe Borzi' <gborzi@ieee.org>
-URL: http://glchess.sourceforge.net
+Summary: 	glChess - A 3D chess interface
+Summary(pl):	glChess - Interfejs 3D do szachów
+Name: 		glchess
+Version: 	0.3.5
+Release: 	1
+License: 	GPL
+Group: 		Amusements/Games
+Group(pl):	Aplikacje/Gry
+Vendor:		Giuseppe Borzi' <gborzi@ieee.org>
+URL: 		http://glchess.sf.net
+Source: 	http://download.sf.net/glchess/%{name}-%{version}.tar.gz
+Source1: 	glchess.menu
+Patch0: 	glchessrc.patch
+Patch1:		patch2.patch
+BuildRequires:	gtk+-devel
+BuildRequires:	gtkglarea-devel
+BuildRoot: 	%{tmpdir}/%{name}-%{version}-root-%(id -u -n)
 
 %description
 glChess is a 3D OpenGL based chess game that interfaces via the Chess Engine
@@ -17,9 +22,16 @@
 Crafty and GNU Chess as AIs. You can also play Human vs. Human, but so far
 not over a netwerk (see TODO).
 
+%description -l pl
+glChess to Trójwymiarowe (3D) szachy w technologii OpenGL. Program u¿ywa 
+Chess Engine Communication Protocol (CECP) autorstwa Tim'a Mann'a. Oznacza
+to Crafty b±d¼ GNU Chess jako AI. Mo¿na tak¿e graæ cz³owiek przeciw cz³owiekowi,
+lecz jeszcze nie przez sieæ (zobacz TODO).
+
 %prep
 %setup -q
 %patch0 -p0
+%patch1 -p0
 
 %build
 make
@@ -48,7 +60,15 @@
 /etc/glchessrc
 %{_sysconfdir}/X11/wmconfig/*
 
+%define date    %(echo `LC_ALL="C" date +"%a %b %d %Y"`)
 %changelog
+* %{date} PLD Team 
+All below listed persons can be reached on <cvs_login>@pld.org.pl
+
+$Log$
+Revision 1.1  2001/11/08 19:10:49  fastviper
+- Spec dla PLD
+
 * Mon Jul 9 2001 Giuseppe Borzi' <gborzi@ieee.org>
 - version 0.3.2
 
