diff -Nru glchess-0.4.2/glchessrc glchess-0.4.2-new/glchessrc
--- glchess-0.4.2/glchessrc	Fri Dec  7 19:51:54 2001
+++ glchess-0.4.2-new/glchessrc	Sun Dec  9 01:35:49 2001
@@ -4,9 +4,9 @@
 is_smooth               1
 is_texture              1
 is_fog                  0
-camera_mode             2
+camera_mode             4
 
-texture_dir             /usr/local/share/games/glchess/textures/
+texture_dir             /usr/share/games/glchess/textures/
 number_colour           1.0 1.0 1.0 1.0
 letter_colour           1.0 1.0 1.0 1.0
 
@@ -14,7 +14,7 @@
 white_piece_colour      0.9 0.9 0.9 1.0
 white_selected_colour   0.9 0.1 0.1 0.8
 white_board_colour      0.78 0.76 0.40 0.5
-white_texture_dir       /usr/local/share/games/glchess/textures/
+white_texture_dir       /usr/share/games/glchess/textures/
 white_piece_texture     white_piece_texture.rgb
 white_selected_texture  selected_piece_texture.rgb
 white_board_texture     white_board_texture.rgb
@@ -23,7 +23,7 @@
 black_piece_colour      0.1 0.1 0.1 1.0
 black_selected_colour   0.9 0.1 0.1 0.8
 black_board_colour      0.47 0.64 0.43 0.5
-black_texture_dir       /usr/local/share/games/glchess/textures/
+black_texture_dir       /usr/share/games/glchess/textures/
 black_piece_texture     black_piece_texture.rgb
 black_selected_texture  selected_piece_texture.rgb
 black_board_texture     black_board_texture.rgb
@@ -32,8 +32,8 @@
 cecp_search_depth       6
 cecp_ai_player          BLACK
 
-path_for_crafty	        ---
-path_for_gnuchessx	---
+path_for_crafty	        /dont/change/that/unless/you/have/crafty
+path_for_gnuchessx	/usr/bin/gnuchess
 
 debug                   1
 beep_signal             1
